src/Elastic.Apm/Helpers/PropertyFetcher.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Elastic.Apm.Helpers
+{
+	internal class PropertyFetcher
+	{
+		private readonly string _propertyName;
+		private PropertyFetch _innerFetcher;
+
+		public PropertyFetcher(string propertyName)
+		{
+			if (string.IsNullOrWhiteSpace(propertyName))
+				throw new ArgumentException("The value must be non-empty, non-null or non-whitespace", nameof(propertyName));
+
+			_propertyName = propertyName;
+		}
+
+		public virtual object Fetch(object obj)
+		{
+			if (_innerFetcher == null)
+			{
+				var type = obj.GetType().GetTypeInfo();
+				var property = type.DeclaredProperties.FirstOrDefault(p =>
+					string.Equals(p.Name, _propertyName, StringComparison.InvariantCultureIgnoreCase));
```

Comment:
I think [the recommendation](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings) is to use `StringComparison.OrdinalIgnoreCase` to compare "symbols":
> Use the non-linguistic StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase values instead of string operations based on CultureInfo.InvariantCulture when the comparison is linguistically irrelevant (symbolic, for example).

Reply:
Thanks for pointing it. Fixed.